app/controllers/api/functions.php

```diff
@@ -60,36 +100,107 @@
     ->label('sdk.response.model', Response::MODEL_FUNCTION)
     ->param('functionId', '', new CustomId(), 'Function ID. Choose your own unique ID or pass the string `ID.unique()` to auto generate it. Valid chars are a-z, A-Z, 0-9, period, hyphen, and underscore. Can\'t start with a special char. Max length is 36 chars.')
     ->param('name', '', new Text(128), 'Function name. Max length: 128 chars.')
-    ->param('execute', [], new Roles(APP_LIMIT_ARRAY_PARAMS_SIZE), 'An array of strings with execution roles. By default no user is granted with any execute permissions. [learn more about permissions](https://appwrite.io/docs/permissions). Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' roles are allowed, each 64 characters long.')
     ->param('runtime', '', new WhiteList(array_keys(Config::getParam('runtimes')), true), 'Execution runtime.')
+    ->param('execute', [], new Roles(APP_LIMIT_ARRAY_PARAMS_SIZE), 'An array of strings with execution roles. By default no user is granted with any execute permissions. [learn more about permissions](https://appwrite.io/docs/permissions). Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' roles are allowed, each 64 characters long.')
     ->param('events', [], new ArrayList(new ValidatorEvent(), APP_LIMIT_ARRAY_PARAMS_SIZE), 'Events list. Maximum of ' . APP_LIMIT_ARRAY_PARAMS_SIZE . ' events are allowed.', true)
     ->param('schedule', '', new Cron(), 'Schedule CRON syntax.', true)
     ->param('timeout', 15, new Range(1, (int) App::getEnv('_APP_FUNCTIONS_TIMEOUT', 900)), 'Function maximum execution time in seconds.', true)
     ->param('enabled', true, new Boolean(), 'Is function enabled?', true)
+    ->param('logging', true, new Boolean(), 'Do executions get logged?', true)
+    ->param('entrypoint', '', new Text('1028'), 'Entrypoint File.', true)
+    ->param('buildCommand', '', new Text('1028'), 'Build Command.', true)
+    ->param('installCommand', '', new Text('1028'), 'Install Command.', true)
+    ->param('vcsInstallationId', '', new Text(128), 'Appwrite Installation ID for vcs deployment.', true)
+    ->param('vcsRepositoryId', '', new Text(128), 'Repository ID of the repo linked to the function', true)
+    ->param('vcsBranch', '', new Text(128), 'Production branch for the repo linked to the function', true)
+    ->param('vcsSilentMode', false, new Boolean(), 'Is VCS connection in silent mode for the repo linked to the function?', true)
+    ->param('vcsRootDirectory', '', new Text(128), 'Path to function code in the linked repo', true)
+    ->param('templateRepositoryName', '', new Text(128), 'Repository name of the template', true)
```

Comment:
Just an idea. Could we reuse the vcs params for the template as well?

Reply:
Not really. Tempalte could be in `appwrite/templates` in folder `node/starter` on branch `main`.
You might want to add function to `meldiron/my-website` in folder  `functions/send-mail` on branch `dev`.

Reply:
After sync discussion, we made soem changes to those params. Keeping the logic, just changing names.